Tests/IGListAdapterTests.m

```diff
@@ -751,13 +751,36 @@ - (void)test_whenScrollVerticallyToBottom_withContentInsets_thatBottomFlushWithC
     self.collectionView.contentInset = UIEdgeInsetsMake(0, 0, 100, 0);
     [self.collectionView layoutIfNeeded];
     [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionVertical scrollPosition:UICollectionViewScrollPositionBottom animated:NO];
-    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900);
+    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900 + 100);
 
     // top 50, bottom 100
     self.collectionView.contentInset = UIEdgeInsetsMake(50, 0, 100, 0);
     [self.collectionView layoutIfNeeded];
     [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionVertical scrollPosition:UICollectionViewScrollPositionBottom animated:NO];
-    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900);
+    IGAssertEqualPoint([self.collectionView contentOffset], 0, 900 + 100);
+    
+    UICollectionViewLayout *layout = self.collectionView.collectionViewLayout;
+    if ([layout isKindOfClass:[UICollectionViewFlowLayout class]]) {
+        [(UICollectionViewFlowLayout *)layout setScrollDirection:UICollectionViewScrollDirectionHorizontal];
+        [layout invalidateLayout];
+        
+        // left 100
+        self.collectionView.contentInset = UIEdgeInsetsMake(0, 100, 0, 0);
+        [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionHorizontal scrollPosition:UICollectionViewScrollPositionRight animated:NO];
+        IGAssertEqualPoint([self.collectionView contentOffset], 900, 0);
+        
+        // right 100
+        self.collectionView.contentInset = UIEdgeInsetsMake(0, 0, 0, 100);
+        [self.collectionView layoutIfNeeded];
+        [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionHorizontal scrollPosition:UICollectionViewScrollPositionRight animated:NO];
+        IGAssertEqualPoint([self.collectionView contentOffset], 900 + 100, 0);
+        
+        // left 50, right 100
+        self.collectionView.contentInset = UIEdgeInsetsMake(0, 50, 0, 100);
+        [self.collectionView layoutIfNeeded];
+        [self.adapter scrollToObject:@100 supplementaryKinds:nil scrollDirection:UICollectionViewScrollDirectionHorizontal scrollPosition:UICollectionViewScrollPositionRight animated:NO];
+        IGAssertEqualPoint([self.collectionView contentOffset], 900 + 100, 0);
+    }
```

Comment:
Love tests!!

Can we move this to its own unit test though? That way if it starts failing we can ID the issue faster.

Reply:
fixed in 7ed9b9757140f235b5dc9751d94b394b12536033